sim/warlock/spell_damage_helper_function.go

```diff
@@ -0,0 +1,108 @@
+package warlock
+
+import (
+	"github.com/wowsims/wotlk/sim/core"
+	"github.com/wowsims/wotlk/sim/core/proto"
+)
+
+func (warlock *Warlock) spellDamageMultiplierHelper(sim *core.Simulation, spell *core.Spell, spellEffect *core.SpellEffect) float64 {

```

Comment:
This function should not be called from OnInit(). Almost all of this logic is static and should be precomputed. The only part that is not static is the execute multiplier, which can be split off into its own effect.

I think a better way to go would be to have a `func (warlock *Warlock) sharedAdditiveMultipliers() float64` function which handles ONLY the generic additive effects, and a  `func (warlock *Warlock) sharedMultiplicativeMultipliers() float64` which handles ONLY the generic multiplicative effects. Then in each of your spells, do something like this:

`DamageMultuplier: (warlock.sharedAdditiveMultipliers() + 0.05*float64(warlock.Talents.ImprovedShadowbolt)) * warlock.sharedMultiplicativeMultipliers() * xxx,`

This will clean up your code a lot and also make it a lot faster.

Reply:
Alright I will do that. Just a question though, about performance. When a function with an "if condition" is called, if the if is false, does the code still try to calculate what's inside the if or is it just ignored and does not affect performance?